tailcfg/tailcfg.go

```diff
@@ -221,9 +221,12 @@ type Node struct {
 	// e.g. "host.tail-scale.ts.net."
 	Name string
 
-	// User is the user who created the node. If ACL tags are in
-	// use for the node then it doesn't reflect the ACL identity
-	// that the node is running as.
+	// User represents the owner of an untagged node. This field should be
+	// ignored for tagged nodes, as the value is undefined and may change in the
+	// future.
+	//
+	// To support old clients, the value is used to group all tagged nodes
+	// together for UI purposes and to circumvent isSelf checks for the PeerAPI.
```

Comment:
To support old clients, the same fake UserID is used for all tagged nodes.
That made them them grouped together in the UI. Also, because the fake UserID for
tagged nodes doesn't correspond to any real human's UserID, ...

wait, the peerapi check is:

```
		isSelf:     nm.SelfNode.User == peerNode.User,
```

Which means this Node.User is correct (a human) for the self node but a fake non-user UserID for all the peers' Nodes.  So the definition of this field changes depending on not only the taggedness but also its parent container?